src/codec.rs

```diff
@@ -147,6 +147,43 @@ impl Codec for u64 {
     }
 }
 
+impl Codec for String {
```

Comment:
No particular objection but I'm curious where this would be needed since we don't use `String`s in any of the MLS structs.

Reply:
Well as soon as we want to send real messages we have strings. Also names are strings. This doesn't have to be handled here but if we want to treat all encoding and decoding the same way this has to go here. We should rethink encoding later anyway then we should be able to move this somewhere else.

Reply:
Sure, we can keep this here for now. In general I'd argue that since the protocol draft only mentions octet strings, anything else should be on the application layer and not in the library.